src/MCPServer/tests/test_package.py

```diff
@@ -123,6 +123,107 @@ def test_transfer_get_or_create_from_db_path_with_uuid(tmp_path):
+        ),
+        "filegrpuse": "original",
+        "transfer_id": transfer_uuid,
+    }
+    models.File.objects.create(**kwargs)
+    assert models.File.objects.filter(transfer_id=str(transfer_uuid)).count() == 1
+
+    # Add a new file to the file-system, e.g. to simulate normalization for
+    # preservation adding a new object.
+    new_file = objects_path / "new_file.txt"
+    new_file.touch()
+
+    # One file will be returned from the database  with a UUID, another from
+    # the filesystem without a UUID.
+    for file_count, file_info in enumerate(transfer.files(None, None, "/objects"), 1):
+        assert "%fileUUID%" in file_info
+        assert "%fileGrpUse%" in file_info
+        assert "%relativeLocation%" in file_info
+        if file_info.get("%fileUUID%") != "None":
+            continue
+        assert file_info.get("%relativeLocation%") == str(new_file)
+        file_path = "".join(
+            [
+                transfer.REPLACEMENT_PATH_STRING,
+                "/objects",
+                file_info.get("%relativeLocation%").split("objects")[1],
+            ]
+        )
+        kwargs = {
+            "uuid": uuid.uuid4(),
+            "currentlocation": file_path,
+            "filegrpuse": "original",
+            "transfer_id": transfer_uuid,
+        }
+        models.File.objects.create(**kwargs)
+    assert (
+        file_count == 2
+    ), "Database and file objects were not returned by the generator"
+    assert models.File.objects.filter(transfer_id=str(transfer_uuid)).count() == 2
+
+    # Simulate an additional file object being added later on in the transfer
+    # in a sub directory of the objects folder, e.g. transcribe contents.
+    sub_dir = objects_path / "subdir"
+    sub_dir.mkdir()
+    new_file = sub_dir / "another_new_file.txt"
+    new_file.touch()
+    for file_count, file_info in enumerate(transfer.files(None, None, "/objects"), 1):
+        file_path = "".join(
+            [
+                transfer.REPLACEMENT_PATH_STRING,
+                "/objects",
+                file_info.get("%relativeLocation%").split("objects")[1],
+            ]
+        )
+        if file_info.get("%fileUUID%") != "None":
+            continue
+        kwargs = {
+            "uuid": uuid.uuid4(),
+            "currentlocation": file_path,
+            "filegrpuse": "original",
+            "transfer_id": transfer_uuid,
+        }
+        models.File.objects.create(**kwargs)
+    assert (
+        file_count == 3
+    ), "Database and file objects were not returned by the generator"
+    assert models.File.objects.filter(transfer_id=str(transfer_uuid)).count() == 3
+
+    # Now the database is updated, we will still have the same file count, but
+    # all objects will be returned from the database and we will have uuids.
+    for file_count, file_info in enumerate(transfer.files(None, None, "/objects"), 1):
+        if file_info.get("%fileUUID%") == "None":
+            print(file_info)
```

Comment:
Forgot to add in the review: this `print` could be avoided by adding the `file_info` in the assert message of the next line.

Reply:
Oh, this is just testing detritus, but it's a good point, some debug info here is always useful, so have added it to the assert message.